src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -245,11 +249,62 @@ void QueryToOperatorTransformer::Visit(parser::InsertStatement *op) {
     op->select->Accept(this);
     insert_expr->PushChild(output_expr_);
     output_expr_ = insert_expr;
-  } else {
-    auto insert_expr = std::make_shared<OperatorExpression>(
-        LogicalInsert::make(target_table, &op->columns, &op->insert_values));
-    output_expr_ = insert_expr;
+    return;
+  }
+  // column_objects represents the columns for the current table as defined in
+  // its schema
+  auto column_objects = target_table->GetColumnObjects();
+  // INSERT INTO table_name VALUES (val1, val2, ...), (val_a, val_b, ...), ...
+  if (op->columns.empty()) {
+    for (uint32_t tuple_idx = 0; tuple_idx != op->insert_values.size(); ++tuple_idx) {
```

Comment:
Can you use a range-for loop with a `const` reference variable? That'll clean up the visual noise.